Merge sorted produces wrong result for lexical categorical

### Checks

- [x] I have checked that this issue has not already been reported.
- [x] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python
import polars as pl
from polars.testing import assert_series_equal

s = pl.Series('a', ["c1", "c0"], pl.Categorical(ordering = "lexical"))

df = pl.DataFrame([s.sort()])

merge_sorted = df.lazy().merge_sorted(df.lazy(), "a").collect().get_column("a")
append_sorted = s.append(s).sort()

assert_series_equal(merge_sorted, append_sorted)
```


### Log output

```shell
Traceback (most recent call last):
  File "/home/johndoe/Projects/polars/bug.py", line 14, in <module>
    test_merge_sorted_self_parametric(pl.Series('a', ["c1", "c0"], pl.Categorical(ordering = "lexical")))
  File "/home/johndoe/Projects/polars/bug.py", line 12, in test_merge_sorted_self_parametric
    assert_series_equal(merge_sorted, append_sorted)
  File "/home/johndoe/Projects/polars/py-polars/polars/_utils/deprecation.py", line 92, in wrapper
    return function(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/johndoe/Projects/polars/py-polars/polars/testing/asserts/series.py", line 117, in assert_series_equal
    _assert_series_values_equal(
  File "/home/johndoe/Projects/polars/py-polars/polars/testing/asserts/series.py", line 197, in _assert_series_values_equal
    raise_assertion_error(
  File "/home/johndoe/Projects/polars/py-polars/polars/testing/asserts/utils.py", line 12, in raise_assertion_error
    raise AssertionError(msg) from cause
AssertionError: Series are different (exact value mismatch)
[left]:  ['c0', 'c1', 'c0', 'c1']
[right]: ['c0', 'c0', 'c1', 'c1']
```

### Issue description

-

### Expected behavior

-

### Installed versions

<details>

```
Replace this line with the output of pl.show_versions(). Leave the backticks in place.
```

</details>

--- a/crates/polars-ops/src/frame/join/merge_sorted.rs
+++ b/crates/polars-ops/src/frame/join/merge_sorted.rs
@@ -146,7 +146,7 @@ fn series_to_merge_indicator(lhs: &Series, rhs: &Series) -> PolarsResult<Vec<boo
     if lhs.dtype().is_categorical() {
         let lhs_ca = lhs.categorical().unwrap();
         if lhs_ca.uses_lexical_ordering() {
-            let rhs_ca = rhs.categorical().unwrap();     
+            let rhs_ca = rhs.categorical().unwrap();
             let out = get_merge_indicator(lhs_ca.iter_str(), rhs_ca.iter_str());
             return Ok(out);
         }

--- a/crates/polars-ops/src/frame/join/merge_sorted.rs
+++ b/crates/polars-ops/src/frame/join/merge_sorted.rs
@@ -143,6 +143,15 @@ where
 }
 
 fn series_to_merge_indicator(lhs: &Series, rhs: &Series) -> PolarsResult<Vec<bool>> {
+    if lhs.dtype().is_categorical() {
+        let lhs_ca = lhs.categorical().unwrap();
+        if lhs_ca.uses_lexical_ordering() {
+            let rhs_ca = rhs.categorical().unwrap();     
+            let out = get_merge_indicator(lhs_ca.iter_str(), rhs_ca.iter_str());
+            return Ok(out);
+        }
+    }
+
     let lhs_s = lhs.to_physical_repr().into_owned();
     let rhs_s = rhs.to_physical_repr().into_owned();
 

--- a/py-polars/tests/unit/operations/test_merge_sorted.py
+++ b/py-polars/tests/unit/operations/test_merge_sorted.py
@@ -75,6 +75,15 @@ def test_merge_sorted_categorical() -> None:
         left.merge_sorted(right, "a")
 
 
+@pytest.mark.may_fail_auto_streaming
+def test_merge_sorted_categorical_lexical() -> None:
+    left = pl.Series("a", ["b", "a"], pl.Categorical("lexical")).sort().to_frame()
+    right = pl.Series("a", ["b", "b", "a"], pl.Categorical("lexical")).sort().to_frame()
+    result = left.merge_sorted(right, "a").get_column("a")
+    expected = left.get_column("a").append(right.get_column("a")).sort()
+    assert_series_equal(result, expected)
+
+
 @pytest.mark.parametrize(
     ("size", "ra"),
     [

